File: Back/LearningApp/TempModels/Question.cs
using System;
using System.Collections.Generic;

namespace LearningApp.TempModels;

public partial class Question
{
    public int QuestionId { get; set; }

    public int QuizId { get; set; }

    public string Type { get; set; } = null!;

    public string Content { get; set; } = null!;

    public int Rank { get; set; }

    public string? Explanation { get; set; }

    public virtual ICollection<QuestionItem> QuestionItems { get; set; } = new List<QuestionItem>();

    public virtual ICollection<QuestionResponse> QuestionResponses { get; set; } = new List<QuestionResponse>();

    public virtual Quiz Quiz { get; set; } = null!;
}

File: Front/Learning-app-v3-student/types/Question.ts
export interface QuestionCreateDto {
    quizId: number;
    type: string;
    content: string;
    rank: number;
    explanation?: string;
    questionItems: QuestionItemCreateDto[];
}

export interface QuestionItemCreateDto {
    questionId: number;
    content?: string;
    isRight?: boolean;
    explanation?: string;
    rightResponse?: string;
}

export interface QuestionDto {
    questionId: number;
    quizId: number;
    type: string;
    content: string;
    rank: number;
    exaplanation?: string;
    questionItems?: QuestionItemDto[];
}

export interface QuestionItemDto {
    questionItemId: number;
    questionId: number;
    content?: string;
    isRight?: boolean;
    explanation?: string;
    rightResponse?: string;
}
